README.md

```diff
@@ -1,55 +1,301 @@
+
+- **Search History**: A list of all searches and interaction data from the users 
+- **Metrics Data**: A collection of recorded metrics from other components which can be used to judge the performance of the engine.
+
+In addition to these two main data stores, we also store **User Feedback**: This will hold any information that the user provides with the expectation of admin action in response. This may include (but is not limited to):
+
+- **Bug Reports**: If the user encounters an issue and would like to report a bug, this report will be stored. The admins who read the report should reach out to the appropriate team to help diagnose and fix the issue
+- **Right to be Forgotten**: A big recurring discussion around this search engine is that the user should be able to have their data scrubbed from the data store. These requests will also be stored here.
+- **General Feedback**: The user will also have the option to provide general feedback, this will also be stored here.
+
+The **Search History** will be stored in an SQLite table. **Metrics Data** is still somewhat up in the air, but we will likely do the same here. **User Feedback** is less structured, and will only be accessed by admin, so it will likely be stored in human readable text files, with the format TBD.
+
+> How & Where are we storing everything?
+
+#### Search History
+
+The search history will be stored as a list of user queries, and the associated action that was taken from them. Each stored query will have the following data:
+
+- **Query ID**: This is a unique identifier for each query, and is stored at request of the Ranking team
+- **Raw Query**: The exact value the user entered for the search
+- **Results**: The list of results which were shown to the user as a result of this query
+- **Clicked**: Which result the user chose
+- **Timestamp**: A timestamp associated with the query
+> Timestamp: When the query started? when it ended? Why do we even need this? I think it is for link analysis webgraph, we don't have to store it, we can just send it to them.
+>
+> Counterargument: We can use this to see how frequently queries occur and how much load is on the search engine
+
+This will be split among various SQLite tables. The exact structure is to be determined.
+
+#### Metrics Data
+
+TBD. Depends on what we want and how we want to show it.
+
+#### User Feedback
+
+Also TBD but maybe just dump into files or something, does not have to be fancy.
+
+### Implementation
+
+Everything the evaluation component does will be started by the receive of a message on a socket. We will have a pool of threads, which wait to be assigned to an incoming request.
+
+> Do we want one port per message type? One port, a threadpool to handle all requests? One thread per request type? Does it even matter if we only have one core and are sharing with DDS?
```

Comment:
Definitely want to prioritize based on message type. 

Importance:
1. Autofill
2. Everything else